exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinProbeTemplate.java

```diff
@@ -47,7 +47,11 @@
 
   private HashJoinBatch outgoingJoinBatch = null;
 
-  private static final int TARGET_RECORDS_PER_BATCH = 4000;
+  private int targetRecordsPerBatch = 4000;
```

Comment:
I would add a comment here about when this value will be mutated as we are moving it away from immutability, and most of the other operators currently have this as an immutable value.